### PR TITLE
HybridFactorGraph emplace variants

### PR DESCRIPTION
**Pull request by dellaert**
emplace variants, automatically deduce the right graph.

Unfortunately emplace variants don't deal well with {}, but it works with auto.

### CODE REVIEW

**Comment by dellaert** [comment]
Review for @ProfFan , cc @varunagrawal

File: gtsam/hybrid/HybridFactorGraph.cpp
Comment (ProfFan) [inline gtsam/hybrid/HybridFactorGraph.cpp:38]: const

File: gtsam/hybrid/HybridFactorGraph.h
Comment (ProfFan) [inline gtsam/hybrid/HybridFactorGraph.h:105]: I agree, since I just looked at GFG and NFG and they all use `shared_ptr` internally.

**Review by ProfFan — APPROVED** [review]

**Comment by varunagrawal** [comment]
I'll wait for this PR to merge before making any updates since I need some of the changes here. :slightly_smiling_face:

File: gtsam/hybrid/HybridFactorGraph.cpp
Comment (dellaert) [reply gtsam/hybrid/HybridFactorGraph.cpp:38]: not even sure why we have these intermediate variables. Feel free to move them in place after I merge.

**Comment by dellaert** [comment]
merged before checks finish. I think the arg forwarding might fail on some platforms, but we can handle in the linearize PR

**Pull request merged**
